CMakeLists.txt

```diff
@@ -500,12 +525,16 @@ if (BUILD_TESTING)
     endif()
     message(STATUS "Running tests with environment: ${UNIT_TEST_ENVS}")
 
+    ############################################################################
+    ############################ build unit tests ##############################
+    ############################################################################
+
     file(GLOB UNITTESTS_SRC "tests/unit/*.c")
     foreach(test_case ${UNITTESTS_SRC})
-        string(REGEX REPLACE ".+\\/(.+)\\.c" "\\1" test_case_name ${test_case})
+        # NAME_WE: name without extension
+        get_filename_component(test_case_name ${test_case} NAME_WE)
 
         add_executable(${test_case_name} ${test_case})
-        target_include_directories(${test_case_name} PRIVATE ./)
         target_link_libraries(${test_case_name} PRIVATE testss2n)
```

Comment:
Any idea what these target_include_directories are trying to do?

Reply:
`target_include_directories` is used to indicate the header file locations that are necessary for that specific target. 

This is an odd directive to use for an individual test binary, because those test binaries don't have associated header files. 

I think the generic thought was "test binary needs access to the test headers" but those test headers are already publicly associated with `testss2n`. So when `testss2n` is linked to the test binary, the test binary will also get the test headers.
https://github.com/aws/s2n-tls/blob/24674164b0353662fd02f792f0c53a786b7720bd/CMakeLists.txt#L453